internal/mcpproxy/handlers.go

```diff
@@ -540,6 +540,14 @@ func (m *MCPProxy) handleToolCallRequest(ctx context.Context, s *session, w http
 		return fmt.Errorf("%w: %s", errInvalidToolName, toolName)
 	}
 
+	// Enforce authentication if required by the route.
+	if route.authorization != nil {
+		if !m.authorizeRequest(route.authorization, headers, backendName, toolName, p.Arguments) {
+			onErrorResponse(w, http.StatusForbidden, "authorization failed")
```

Comment:
According to the new MCP spec, the MCP Gateway should include the missing scopes in the WWW-Authenticate header of the 403 response when authorization fails.

However, adding theses scopes is tricky because our authorization model is more flexible than the one defined in the MCP spec. For example, there are cases where we simply don't have meaningful scopes to return:
* No rules with matching tools are found for the target
* The rule is based on other claims in the source (not scopes), so there's no corresponding scope we can include in the response.